### DISCUSSION TITLE
Getting Type Error: '{ ttl: number; swr: number; }' is not assignable to type 'never'.ts(2322)

### DISCUSSION THREAD

**Discussion by kulterryan** [prisma/prisma#23719] [Q&A]
### Bug description

I'm getting type error in Next.js 14 with Prisma Accelerate, not able to cache data properly from accelerate.

![Screenshot 2024-03-31 at 16 11 52](https://github.com/prisma/prisma/assets/36154235/0bc19da2-e038-414a-a227-8181eb22a0a3)

### How to reproduce

Create a function with findMany method.

```
 const fetchBusiness = await prisma.business.findMany({
    orderBy: {
      name: 'asc'
    },
    cacheStrategy: {
      ttl: 3_600,
      swr: 500,
    },
  });
```

### Expected behavior

No Type error.

### Prisma information

// Add your schema.prisma
```
model Business {
  id              String            @id @default(auto()) @map("_id") @db.ObjectId
  name            String
  description     String
  website         String
....
}
```
// Add your code using Prisma Client
```
import { PrismaClient } from '@prisma/client/edge';
import { withAccelerate } from '@prisma/extension-accelerate';

const globalForPrisma = global as unknown as { prisma: PrismaClient };

export const prisma =
  globalForPrisma.prisma ||
  new PrismaClient({
    log: ['error', 'warn'],
  }).$extends(withAccelerate());

if (process.env.NODE_ENV !== 'production') globalForPrisma.prisma = prisma;
```

### Environment & setup

- OS: MacOS
- Database: MongoDB
- Node.js version: v20.11.0

### Prisma Version

```
prisma                  : 5.11.0
@prisma/client          : 5.11.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine efd2449663b3d73d637ea1fd226bafbcf45b3102 (at node_modules/.pnpm/@prisma+engines@5.11.0/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli efd2449663b3d73d637ea1fd226bafbcf45b3102 (at node_modules/.pnpm/@prisma+engines@5.11.0/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.11.0-15.efd2449663b3d73d637ea1fd226bafbcf45b3102
Default Engines Hash    : efd2449663b3d73d637ea1fd226bafbcf45b3102
Studio                  : 0.499.0
Preview Features        : fullTextSearch
```

**Comment by nurul3101** [comment]
Hey @kulterryan 

Have you invoked `npx prisma generate --no-engine` command?

Can you also try restarting your TypeScript server in VS Code (Assuming you are using VS Code)

**Reply by kulterryan** [reply]
Yes, I did both but nothing worked.

**Reply by nurul3101** [reply]
Okay, Is your repository open source? If yes, could you share the link?
I would like to reproduce this on my end and check.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by jcdk** [comment]
@kulterryan were you able to resolve this issue? The issue just started happening to me.

**Comment by jcdk** [comment]
I was able to fix this by changing:

```
import { PrismaClient } from "@prisma/client/edge";
import { withAccelerate } from "@prisma/extension-accelerate";

const globalForPrisma = globalThis as unknown as { prisma: PrismaClient };

export const db = globalForPrisma.prisma || new PrismaClient().$extends(withAccelerate());

if (process.env.NODE_ENV !== 'production') globalForPrisma.prisma = db;

```

to:
```
import { PrismaClient } from "@prisma/client/edge";
import { withAccelerate } from "@prisma/extension-accelerate";

export const db = new PrismaClient().$extends(withAccelerate());
```

**Reply by kulterryan** [reply]
Yes, updating to latest version resolved the issue for me.

**Comment by Skb3611** [comment]
is this solved? im facing the same issue
